File: src/main/kotlin/com/kyfstore/tradeMC/TradeMCPluginLoader.java
package com.kyfstore.tradeMC;

import io.papermc.paper.plugin.loader.PluginClasspathBuilder;
import io.papermc.paper.plugin.loader.PluginLoader;
import io.papermc.paper.plugin.loader.library.impl.MavenLibraryResolver;
import org.eclipse.aether.artifact.DefaultArtifact;
import org.eclipse.aether.graph.Dependency;
import org.eclipse.aether.repository.RemoteRepository;

class TradeMCPluginLoader implements PluginLoader {
    @Override
    public void classloader(PluginClasspathBuilder pluginClasspathBuilder)
    {
        MavenLibraryResolver resolver = new MavenLibraryResolver();
        resolver.addRepository(new RemoteRepository.Builder("xenondevs", "default", "https://repo.xenondevs.xyz/releases/").build());
        resolver.addDependency(new Dependency(new DefaultArtifact("xyz.xenondevs.invui:invui:pom:VERSION"), null));
        pluginClasspathBuilder.addLibrary(resolver);
    }
}

File: src/main/kotlin/com/kyfstore/tradeMC/TradeMCPluginLoader.kt
package com.kyfstore.tradeMC

import io.papermc.paper.plugin.loader.PluginClasspathBuilder
import io.papermc.paper.plugin.loader.PluginLoader
import io.papermc.paper.plugin.loader.library.impl.MavenLibraryResolver
import org.eclipse.aether.artifact.DefaultArtifact
import org.eclipse.aether.graph.Dependency
import org.eclipse.aether.repository.RemoteRepository

class TradeMCPluginLoader: PluginLoader {
    override fun classloader(pluginClasspathBuilder: PluginClasspathBuilder) {
        val resolver = MavenLibraryResolver()
        resolver.addRepository(RemoteRepository.Builder("xenondevs", "default", "https://repo.xenondevs.xyz/releases/").build())
        resolver.addRepository(RemoteRepository.Builder("mavenCentral", "default", "https://oss.sonatype.org/content/groups/public/").build())
        resolver.addDependency(Dependency(DefaultArtifact("xyz.xenondevs.invui:invui:pom:1.44"), null))
        resolver.addDependency(Dependency(DefaultArtifact("dev.jorel:commandapi-bukkit-shade-mojang-mapped:9.7.0"), null))
        pluginClasspathBuilder.addLibrary(resolver)
    }
}